src/List/ItemList.js

```diff
@@ -22,22 +24,48 @@ const Items = props => {
     }
   }, [token, setdbItems, dbItems, props]);
 
-  //This will update an item to include a purchase date
-  const handleChange = e => {
-    let datePurchased = new Date();
-    let db = firebase.fb.firestore();
-    db.collection(token)
-      .doc(e.target.value)
-      .update({ datePurchased });
-    console.log(dbItems);
+  // Checks if a purchase date already exists - if not, creates a purchased date and increments # of purchases - if so, also sets the most recent purchase estimate, the most recent purchase interval and the calculated date of the next purchase
+  const handleChange = (e, item) => {
+    if (item.datePurchased) {
+      let lastEstimate;
+      item.nextPurchaseDate
+        ? (lastEstimate = item.nextPurchaseDate)
+        : (lastEstimate = item.frequency);
+      let lastDatePurchased = item.datePurchased;
+      let datePurchased = new Date();
+      let datePurchasedInSeconds = Math.floor(datePurchased.getTime() / 1000);
+      let latestInterval = Math.floor(
+        (datePurchasedInSeconds - lastDatePurchased.seconds) / HOURS24
+      );
+      let db = firebase.fb.firestore();
+      let nextPurchaseDate = calculateEstimate(
+        item.lastEstimate,
+        latestInterval,
+        item.numOfPurchases
+      );
+      db.collection(token)
+        .doc(e.target.value)
+        .update({
+          datePurchased,
+          numOfPurchases: item.numOfPurchases + 1,
+          latestInterval,
+          lastEstimate,
+          nextPurchaseDate
+        });
+    } else {
+      let datePurchased = new Date();
+      let db = firebase.fb.firestore();
+      db.collection(token)
+        .doc(e.target.value)
+        .update({ datePurchased, numOfPurchases: item.numOfPurchases + 1 });
```

Comment:
If there is no `datePurchased`, will `numOfPurchases` always be 0, so we always send 1 here?